modules/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java

```diff
@@ -490,7 +484,7 @@ public static void startPendingHttpConnectors(BallerinaHttpServerConnector httpS
         MessageDataSource outboundMessageSource = HttpUtil.getMessageDataSource(httpMessageStruct);
         HttpResponseStatusFuture outboundResponseStatusFuture = sendOutboundResponse(requestMessage, responseMessage);
         if (outboundMessageSource != null) {
-            outboundMessageSource.serializeData();
+            outboundMessageSource.serializeData((OutputStream) httpMessageStruct.getNativeData(MESSAGE_OUTPUT_STREAM));
```

Comment:
Instead of getting output stream from native data and casting it, can't we create the output stream here itself?